### ISSUE TITLE
Make it work without vim-addon-manager

### ISSUE THREAD

**Issue by andreypopp** [jlc/envim#2]
Some users already have preferences regarding how to manage their vim configuration and vim-addon-manager isn't the only available option there. I use pathogen and can't get Envim work with my setup, first simply because I have no vim-addon-manager installed and even after I installed it in pathogen's managed bundle dir Envim still doesn't work for me.

I think it would be great if Envim would make no assumption about what vim plugin manager is in use.

Thank you for Envim!

**Comment by jlc** [comment]
True, envim should be independent from how users are managing their configurations.

vim-addon-manager requirement was set in order to focus on the main features (like an omni-completion menu filled by ensime :))
However, now it is important to work on installation and usability issues, and your comment is greatly appreciated.

I'll update this issue as soon as a better solution is implemented.

**Comment by DavidGamba** [comment]
I am using Vundle. 
Added the following lines in my .vimrc and then ran :BundleUpdate 

" Ensime
Bundle "git://github.com/aemoncannon/ensime.git"
Bundle "git://github.com/jlc/envim.git"
Bundle "git://github.com/jlc/ensime-common.git"
Bundle "git://github.com/jlc/vim-async-beans.git"
Bundle "git://github.com/MarcWeber/vim-addon-signs.git"
Bundle "git://github.com/jlc/vim-addon-async.git"

I got a whole bunch of errors due to python support so had to recompile vim to add python support:
./configure --with-compiledby="David Gamba davidgamba@mydomain.com" \
    --with-features=huge \
    --enable-gui=auto \
    --with-x \
    --enable-rubyinterp \
    --enable-perlinterp \
    --enable-pythoninterp --with-python-config-dir=/usr/lib/python2.7/config \
    --enable-fontset \
    --enable-cscope \
    --enable-gtk2-check \
    --enable-gnome-check

I am trying to test it with the play framework so my .ensime file was generated by the framework:
http://www.playframework.com/documentation/2.1.x/IDE

Things aren't working for me, I wonder if it is due to the fact that I am using ensime from master and I am also using scala 2.10.

Will update here if I have any luck.

**Comment by DavidGamba** [comment]
Cloned ensime into a local dir and renamed scala-2.9 to master (To bypass Vundle's inability to select branches). It is still not working for me:

 Error detected while processing function envim#onCursorMoved ... NameError: name 'Envim' is not defined.

Lame!!! for now this is a no go for me, I like the ease of usage of Vundle and already spent a couple hours playing around. 

If you have any hints onto what I should try let me know, I am happy to help but don't know where to start.

**Comment by DavidGamba** [comment]
I didn't noticed that vam commands where actually part of the envim code:

```
not found in 'runtimepath': "autoload/vam.vim"
line   36:
E117: Unknown function: vam#DefaultPluginDirFromName
E116: Invalid arguments for function get(g:envim, 'path-ensime-common', vam#DefaultPluginDirFromName('ensime-common'))
E15: Invalid expression: get(g:envim, 'path-ensime-common', vam#DefaultPluginDirFromName('ensime-common'))
line   75:
Traceback (most recent call last):
  File "<string>", line 7, in <module>
vim.error: Vim:E716: Key not present in Dictionary: path-ensime
finished sourcing /home/david/.vim/bundle/envim/autoload/envim.vim
continuing in VimLeave Auto commands for "*"
```

Output of grep:

```
bundle/vim-addon-signs/autoload/vim_addon_signs.vim :" exec vam#DefineAndBind('s:c','g:vim_addon_signs','{}')
bundle/vim-addon-signs/plugin/vim-addon-signs.vim :" exec vam#DefineAndBind('s:c','g:vim_addon_signs','{}')
bundle/vim-addon-async/autoload/async.vim :elseif vam#IsPluginInstalled('vim-async-beans')
bundle/envim/README.md :call vam#ActivateAddons(plugins,{'auto_install' : 0})
bundle/envim/autoload/envim.vim :if !vam#IsPluginInstalled('ensime')
bundle/envim/autoload/envim.vim :if !vam#IsPluginInstalled('ensime-common')
bundle/envim/autoload/envim.vim :let g:envim['path-ensime'] = get(g:envim, 'path-ensime', vam#DefaultPluginDirFromName('ensime'))
bundle/envim/autoload/envim.vim :let g:envim['path-ensime-common'] = get(g:envim, 'path-ensime-common', vam#DefaultPluginDirFromName('ensime-common'))
bundle/envim/autoload/envim.vim :if !vam#IsPluginInstalled('vim-async-beans')
```

Sorry for all the spamming.

**Comment by jlc** [comment]
Hi @DavidGamba,

Thanks for your interest on Envim and experimentations!
Spamming is good when containing pertinent information!

It is true that Envim should be package manager agnostic.

Vim is a wonderful editor, and its key binding follow me everywhere, from shell, tmux, emails, to editor.

However, while developing envim, I realised that:
1) its scripting language doesn't match my expectations (even with Python in the middle), and
2) its chances to become an asynchronous friendly editor are very small (even if a lot of people asking for it), and every workarounds (such as https://github.com/jlc/vim-async-beans) or patches (such as https://github.com/jlc/vim) will always be workarounds and external patches.

I found that emacs provides a strong operating system with native async support, plugins are written in lisp (good old pure functional language), and most notably Evil which provide excellent vim binding, and, of course, Ensime.

Therefore, I didn't continue the development of envim (this will be said in the README very soon).

However, I would be pleased to help and support as much as possible anyone who would like to keep on maintaining this project!

The first envim contributor, @megaannum, developed vimside in order to use ensime within vim.
The project seems to be actively developed, you should have a look!
https://github.com/megaannum/vimside

With best wishes,

Jeanluc
